src/chapter08/exercises/exercise01.lean

```diff
@@ -6,10 +6,39 @@ change, any whole number of roubles greater than 7 with
 banknotes of value 3 roubles and 5 roubles
 -/
 
+
 example (n : ℕ) (hn : 7 < n) : ∃ a b : ℕ, n = 3 * a + 5 * b :=
 begin
   -- Remark: this looks a little subtle to me, you will have
   -- to think about exactly what statement you want to prove
   -- by induction.
-  sorry
-end
\ No newline at end of file
+  generalize hd : n - 8 = d,
+  have hnd : n = d + 8,
+  { linarith, },
+  rw hnd,
+  clear hnd hd hn n,
+  induction d with k hk,
+  { use 1, use 1, norm_num, },
+  change ∃ a b, k + 1 + 8 = 3 * a + 5 * b,
+  rcases hk with ⟨a, b, hp⟩,
+  by_cases h : 0 < b,
+  { use a+2, use b-1, linarith, },
+  push_neg at h,
+  simp at h,
```

Comment:
This is bad style -- it's a "non-terminal `simp`". You should change it temporarily to `squeeze_simp at h` and then click on "try this" and replace `squeeze_simp at h` with the output. The problem with `simp` is that it is a "living tactic" -- as mathlib changes the behaviour of `simp` changes too, so if we upgrade mathlib then it might break your code, unless you replace it with `simp only...`